File: 1. Two Sum/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  const index = {}
  for (let i = 0; i < nums.length; i++) {
    const matchNum = target - nums[i];
    if (typeof index[matchNum] === 'number') return [index[matchNum], i]
    index[nums[i]] = i;
  }
};

File: 1. Two Sum/solution.kt
class Solution {
  fun twoSum(nums: IntArray, target: Int): IntArray {
    val res = IntArray(2)
    val map = mutableMapOf<Int, Int>()
      
    for ((i, value) in nums.withIndex()) {
      map.get(target - value)?.let { j ->
        res[0] = j
        res[1] = i
        return res
      }      
      map[value] = i
    }
      
    return res
  }
}